src/core/fetchers/segment/segment_queue.ts

```diff
@@ -257,6 +259,12 @@ export default class SegmentQueue<T> extends EventEmitter<ISegmentQueueEvent<T>>
     const { downloadQueue, content, initSegmentInfoRef, currentCanceller } = contentInfo;
 
     const recursivelyRequestSegments = (): void => {
+      if (contentInfo.canStream.getValue() === false) {
+        log.info(
+          "DEBUG ManagedMediaSource: segment fetching postponed because it cannot stream now",
+        );
```

Comment:
I presume the `DEBUG` prefix was here for debugging.

Do we have a namespace for `SegmentQueue` elsewhere in the file? If we do, we can just reuse it instead, if not, we could add one, like SQ.

Reply:
Also, I see where we don't load anymore, but I don't see where we restart loading segments.

Shouldn't the `SegmentQueue` handle both?

Reply:
The code that restart loading segment is already in the segment queue:
          
```js
else if (currentSegmentRequest === null) {
  // There's no request although there are needed segments: start requests
  log.debug(
    "SQ: Media segments now need to be requested. Starting queue.",
    content.adaptation.type,
    segmentQueue.length,
  );
  this._restartMediaSegmentDownloadingQueue(currentContentInfo);
  return;
}
```

Reply:
> The code that restart loading segment is already in the segment queue: [...]

That code is only triggered when the downloadqueue is updated though.

If it isn't updated and `canLoad` goes from `false` to `true`, the awaiting segment requests won't be started.